src/adc.rs

```diff
@@ -639,64 +454,530 @@ impl Default for SampleTime {
 }
 
 /// Implemented for all types that represent ADC channels
-pub trait Channel: EmbeddedHalChannel<ADC, ID = u8> {
-    fn set_sample_time(&mut self, adc: &ADC1, sample_time: SampleTime);
+pub trait Channel<T>: EmbeddedHalChannel<T, ID = u8> {
+    fn set_sample_time(&mut self, adc: &mut T, sample_time: SampleTime);
+}
+
+macro_rules! impl_embedded_hal_channel {
+    ($pin:ty => ($adc_type:ident, $chan:expr)) => {
+        impl EmbeddedHalChannel<pac::$adc_type> for $pin {
+            type ID = u8;
+
+            fn channel() -> Self::ID {
+                $chan
+            }
+        }
+    };
+}
+
+macro_rules! impl_channel {
+    ($pin:ty => ($adc_type:ident, $smpr:ident, $smp:ident, $($min_sample_time:expr)?)) => {
+        impl Channel<pac::$adc_type> for $pin {
+            #[inline]
+            fn set_sample_time(&mut self, adc: &mut pac::$adc_type, sample_time: SampleTime) {
+                $(
+                    let sample_time = $min_sample_time;
```

Comment:
This overwrites the methods input `sample_time` parameter, is this intended?
If so this needs an explaining comment.

Reply:
Yes, it's intended. `Vref` needs a minimum sample time so it overrides the default one.

Reply:
But then why is the argument there? It is essentially unused now right?